wiring/inc/spark_wiring_cellular.h

```diff
@@ -74,6 +74,12 @@ class CellularClass : public NetworkClass
 
     CellularSignal RSSI();
 
+    CellularData dataUsage(void);
+
+    CellularData dataUsage(const CellularData &data_set);
```

Comment:
I wanted to start with data = Cellular.dataUsage() as a way to get the current data usage values.  Then I added a way to set the values, which I added an overloaded function.  It seemed intuitive to pass in a value to set the counters, like creating a `String sweet = String("dude")`.  And even though we can reset the count now with `Cellular.dataUsage(zeroedOutData)` I felt like adding a convenience function just for that was helpful `Cellular.dataUsageReset()` opting to add Reset to the end for the day when we have intellitype autocomplete in the IDE :) Of course we could make everything very explicit:

Cellular.dataUsageGet()
Cellular.dataUsageSet()
Cellular.dataUsageReset()

Which wouldn't be so bad. I often try to keep the typing and number of humps in the camelCase to a minimum if possible.  It also seems unnecessary to spell everything out when we have a lot of functions in our API like Cellular.RSSI() WiFi.localIP() System.deviceID() etc.. I'm totally open to going one way or the other though if we'd like to define signatures for when either approach is best.  Like if an API has set/get functions, make them explicit.

The counter value is saved in the wiring API in a global instance of CellularData, so the object that the user defines in their application.

I chose `CellularData` because the values are fairly specific to this modem, and we have a running theme of types for cellular: CellularSignal, CellularDevice, CellularCredentials.  You can see there is CID element in CellularData which is the PDP Context index for the data in the counters, fairly specific to cellular.  I think we need a generic DataUsage object currently, but could maybe see making one in the future that pulls these specific types into it.

I only gave the ability to set the counters to a specific value because the original AT command has this same functionality.  I'm guessing this might be appropriate for aggregating a running count over a long period of time.  Maybe when a user app wakes up it pulls the values from EEPROM, and then saves them back before sleeping.

When setting the dataUsage counters, the only thing different in the returned value might be the CID element.  It's used as a return value error in the case that the call to the modem failed.  So we can't assume the counter object we passed in is has the same values exactly.  We could set the values by reference and update the CID value that way as well, I just think this interface changes the way the return value is interpreted from the GET function variant.  So coming full circle, I like the simplicity of getting a returned object CellularData from Cellular.dataUsage() that contains all of our counter info, and then to set those values just overloading Cellular.dataUsage(const CellularData&) with the ability to keep our error checking the same way as with the Get function, in the returned object.